pages/k8s/monitoring.md

```diff
@@ -0,0 +1,263 @@
+```
+
+### Retrieve  credentials and login
+
+To open the dashboard in your browser you will need to know the IP address for
+**grafana** and the admin password. These can be retrieved with the following
+commands:
+
+```bash
+juju status --format yaml grafana/0 | grep public-address   
+```
+Will return the accessible IP address for the dashboard.
+
+```bash
+juju run-action --wait grafana/0 get-admin-password
+```
+Will return the password for the user 'admin'
+
+You can now navigate to the website at `http://<your-ip>:3000`  and login with
+the username `admin` and the password you just retrieved.
+
+Once logged in, check out the cluster metric dashboard by clicking the `Home`
+drop-down box and selecting `Kubernetes Metrics (via Prometheus)`:
+
+![grafana dashboard image][grafana-1]
+
+You can also check out the system metrics of the cluster by switching the
+drop-down box to `Node Metrics (via Telegraf):
+
+![grafana dashboard image][grafana-2]
+
+
+## Monitoring with Nagios
+
+**Nagios** ([https://www.nagios.org/][nagios]) is widely used for monitoring
+networks, servers and applications. Using the Nagios Remote Plugin Executor
+(NRPE) on each node, it can monitor your cluster with machine-level detail.
+
+To start, deploy the latest version of the Nagios and NRPE Juju charms:
+
+```bash
+juju deploy nagios --series=xenial
+juju deploy nrpe --series=xenial
+juju expose nagios
+```
+
+Connect **Nagios** to NRPE:
+
+```bash
+juju add-relation nagios nrpe
+```
+
+Now add relations to NRPE for all the  applications  you wish to monitor, for
+example kubernetes-master, kubernetes-worker, etcd, easyrsa, and
+kubeapi-load-balancer.
+
+```bash
+juju add-relation nrpe kubernetes-master
+juju add-relation nrpe kubernetes-worker
+juju add-relation nrpe etcd
+juju add-relation nrpe easyrsa
+juju add-relation nrpe kubeapi-load-balancer
+```
+
+To connect to the Nagios server, you will need its IP address:
+
+```bash
+juju status --format yaml nagios/0 | grep public-address
+```
+
+The default username is `nagiosadmin`. The password is randomly generated at
+install time, and can be retrieved by running:
+
+```bash
+juju ssh nagios/0 sudo cat /var/lib/juju/nagios.passwd
+```
+
+![nagios dashboard image][nagios-1]
+
+### Using an existing Nagios service
+
+If you already have an existing **Nagios** installation, the
+`nrpe-external-master` charm can be used instead.
+
+```bash
+juju deploy nrpe-external-master --series=xenial
+juju configure nrpe-external-master nagios_master= <ip-address-of-nagios>
+```
+
+Once configured, add relations to `nrpe-external-master`  as outlined above.
+
+## Monitoring with **Elasticsearch**
+
+Elasticsearch ([https://www.elastic.co/][elastic]) is a popular monitoring application which
+can be used in conjunction with **CDK**.
+
+### Deploy the required applications
+
+Use Juju to deploy the required applications:
+
+```bash
+juju deploy elasticsearch --series=xenial
```

Comment:
Yo @evilnick, since this isn't merged yet, will you add a constraint here:
```diff
- juju deploy elasticsearch --series=xenial
+ juju deploy elasticsearch --series=xenial --constraints mem=4G
```

We had a recent report of ES not coming up reliably on machines with < 4G ram.

Reply:
Sure! Looking at this again, I need to make a note to update it when we move to bionic as well